Resources/Public/JavaScript/add_to_cart.js

```diff
@@ -0,0 +1,215 @@
+            errorElement.innerHTML = response.messageBody;
+            errorContainer.style.display = null;
+            fadeOut(errorContainer, messageTimeout);
+
+            dispatchCustomEvent(
+                'render-add-to-cart-result-message',
+                {
+                    response: response,
+                    success: false,
+                    element: errorElement,
+                }
+            );
+        }
+    }
+
+    /**
+     * Use CSS for fade-out (is much more performant than JS)
+     * @param element HTML element to fade out
+     * @param messageTimeout time after which the effect starts
+     */
+    function fadeOut(element, messageTimeout) {
+        const transitionTime = 200;
+        element.style.transition = 'opacity ' + transitionTime + 'ms ease';
+
+        // the fade out
+        window.setTimeout(function () {
+            element.style.opacity = 0;
+        }, messageTimeout);
+
+        // reset element after fade out
+        window.setTimeout(function () {
+                element.style.transition = 'unset';
+                element.style.display = 'none';
+                element.style.opacity = 1;
+            },
+            messageTimeout + transitionTime
+        );
+
+        dispatchCustomEvent(
+            'hide-message-block',
+            {
+                element: element
+            }
+        );
+    }
+
+    function updateMiniCart(form, data) {
+        let response = JSON.parse(data);
+        if (response.status !== "200") {
+            return;
+        }
+
+        let count = response.count;
+        let net = response.net;
+        let gross = response.gross;
+
+        let miniCart = document.querySelector('#cart-preview')
+        let countElement = miniCart.querySelector('.cart-preview-count');
+        let netElement = miniCart.querySelector('.net');
+        let grossElement = miniCart.querySelector('.gross');
+        let linkElement = miniCart.querySelector('.checkout-link');
+
+        if (countElement) {
+            countElement.innerHTML = count;
+        }
+        if (netElement) {
+            netElement.innerHTML = net;
+        }
+        if (grossElement) {
+            grossElement.innerHTML = gross;
+        }
+
+        if (linkElement) {
+            if (count > 0) {
+                linkElement.style.display = 'block';
+            } else {
+                linkElement.style.display = 'none';
+            }
+        }
+
+        dispatchCustomEvent(
+            'minicart-was-updated',
+            {
+                count: response.count,
+                net: response.net,
+                gross: response.gross
+            }
+        );
+
+        document.querySelectorAll('form').forEach(function (formElement) {
+            formElement.reset();
+        })
+    }
+
+    function dispatchCustomEvent(name, dataObject) {
+        const customEvent = new CustomEvent(
+            `extcode:{name}`,
```

Comment:
Did you mean?

```suggestion
            `extcode:${name}`,
```

Because otherwise, this line simply renders as string `'extcode:{name}'` in all browsers

Reply:
@koehnlein 
Of course did I miss the $ sign (and copied it over and over again...) Thanks for the hint. I force-pushed an update.